README
aNexter
=======

aNexter for Factory Test, now its open source.

Powerful Tester, need you joining it.

(c) Tody 2012, T-ware Inc.


CHANGES
v1.2.5.9-1 @ 2012/06/25
*) add M6 Support for SD & USB Test...

v1.2.5.8 @ 2012/04/17
*) fix WiFi MAC crash issue
*) fix gyro UI

v1.2.5.7 @ 2012/04/16
*) fix finish issue

v1.2.5.6 @ 2012/03/28
*) fix WiFi automatic connection issue

v1.2.5.5 @ 2012/03/28
*) add remove Pendrive function for RK.
*) change Install location to internal storage.

v1.2.5.4 @ 2012/03/26
*) fix WiFi auto connection support for Android 4.1
*) fix WiFi global vars for some crash issues.
*) add wifi auto connection @ WiFi & BT Mac.
*) remove no Bluetooth show NO Bluetooth Device issue. 
*) change light sensor pass value to 100(lx) for rockchip;

v1.2.5.3 @ 2012/03/13
*) hidden hdmi switch when the platform is rk30
*) hidden vitual keys when the platform is rk30
*) remove docking
*) sdToPc remove docking
*) fix gyro auto pass to suspend issue
*) fix eMMC size over 16G issue for rk30
*) add none mcu condition for rk30
*) add internal compass function
*) add enable WiFi and Bluetooth function
*) change Bluetooth Fail button to enable
*) change Compass Fail button to enable
*) change light sensor tips to red.


v1.2.5.2 @ 2012/03/11
*) fix crash when check WiFi & MAC address
*) fix eMMC storage size issue

v1.2.5.1 @ 2012/12/24
*) add N/A function for WiFi Test

v1.2.5.0 @ 2012/12/21
*) add SD Card Support for TAxCPx
*) add USB support for TAxCPx
*) fix bluetooth when bluetooth absent

v1.2.4.2 @ 2012/10/30
*) bug fix Bluetooth Crash Issue

v1.2.4.1 @ 2012/10/30
*) bug fix HDMI Crash Issue

v1.2.4 @ 2012/08/17
*) change HDMI search Path to /mnt and add HDMI loop function
*) fix HDMI Layout

v1.2.3 @ 2012/08/16
*) add light Sensor function
*) fix U Disk layout issue
*) change standard lux value to 800lx

v1.2.2
*) add new Page for U Disk test, function as Sdcard R/W Test, remove iFilemanager
*) add HDMI video test, app can search Mp4 or 3gp video files random
*) fix LCD RGB Test hang up after suspend

v1.2.0 @ 2012/08/15
*) change WiFi auto-connection function
*) chnage Bluetooth auto searching function

v1.1.0 @ 2012/05/09
*) Changed some test items positions(G-sensor, mp3, record, vibrator, sdrw, brightness, mac, image, hdmi)
*) Changed xml progress
*) Changed Layout progress
*) Removed suspend test
*) Removed ethernet test
*) Added background music @ LCD RGB Test
*) Added source code to github.com(https://Tody-Guo@github.com/Tody-Guo/aNexter.git)

v1.0.9 @ 2012/05/02
*) First release to github.com


src/com/tware/anexter/uDisk.java
package com.tware.anexter;


import java.io.File;
import java.io.FileOutputStream;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

import android.app.Activity;
import android.content.ActivityNotFoundException;
import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.util.Log;
import android.view.KeyEvent;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.Button;
import android.widget.Toast;

public class uDisk extends Activity {
	private final String TAG = "uDisk";
	private Button bPass;
	private Button bFail;
	private Button bNa;
	private String LOG;
	private List<String> uList = new ArrayList<String>();
	
	/** Called when the activity is first created. */
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.udisk);
        this.setTitle("aNexter - U 盘测试");
        
        bPass = (Button)findViewById(R.id.btn_pass);
        bPass.setEnabled(false);
        bPass.setOnClickListener(new OnClickListener(){
        	@Override
			public void onClick(View v)
        	{
        		if (util.isRk())
        			rmPenDrive();
        		
        		Intent i = new Intent();
        		i.putExtra("LOG", LOG + "PASS|");
        		i.setClass(uDisk.this , uHDMI.class);
        		startActivity(i);
        		uDisk.this.finish();
        	}
        });
        
        Button iFile = (Button)findViewById(R.id.btn_Udisk);
        iFile.setOnClickListener(new OnClickListener(){
        	@Override
        	public void onClick(View v)
        	{
        		uList.clear();
        		if (!getUDiskDirs())
        		{
					Toast.makeText(getApplicationContext(), "找不到U盘装置!", Toast.LENGTH_LONG).show();							
					bPass.setEnabled(false);
					return ;
        		}
        		for (int i=0; i< uList.size(); i++)
        		{
            		try {
            			File frw = new File(uList.get(i).toString()+"/uRW.txt");
            			Log.e("U Disk", "RW @ " + uList.get(i).toString());
						frw.createNewFile();
						FileOutputStream wf = new FileOutputStream(frw);
						wf.write("\r\nU Disk Test String from aNexter\r\n(c) Tody 2012, T-ware Inc.\r\n".getBytes());
						wf.flush();
						wf.close();
						frw.delete();
						Log.d("U Disk", "U Disk Read/Write Ok");
						
		        		Intent i1 = new Intent();
		        		i1.putExtra("LOG", LOG + "PASS|");
		        		i1.setClass(uDisk.this , uHDMI.class);
		        		startActivity(i1);
		        		uDisk.this.finish();
		        		
            		}catch (IOException e) {
						e.printStackTrace();
						Toast.makeText(getApplicationContext(), "U盘读写失败!", Toast.LENGTH_LONG).show();							
						bPass.setEnabled(false);
            		}
        		}
        		/* this is for remove Pendrive of TA0CP, NEW ADDED FUNCTION @ 2013/03/28 */
        		if (util.isRk())
        			rmPenDrive();
        	}
        });
       
        bFail = (Button)findViewById(R.id.btn_fail);
        bFail.setOnClickListener(new OnClickListener(){
        	@Override
    		public void onClick(View v)
        	{
        		Intent i = new Intent();
        		i.putExtra("LOG", LOG + "FAIL|");
        		i.setClass(uDisk.this , getResults.class);
        		startActivity(i);
        		uDisk.this.finish();
        	}
        });
        
        bNa = (Button)findViewById(R.id.btn_na);
//        bNa.setVisibility(View.INVISIBLE);
        bNa.setOnClickListener(new OnClickListener(){
        	@Override
    		public void onClick(View v)
        	{
        		Intent i = new Intent();
        		i.putExtra("LOG", LOG + "N/A|");
        		i.setClass(uDisk.this , uHDMI.class);
        		startActivity(i);
        		uDisk.this.finish();
        	}
        });
        
        LOG = this.getIntent().getStringExtra("LOG");
    }
    
    public void rmPenDrive()
    {
        Log.d(TAG, "Send ShuttleUnmount for /mnt/usb_storage Intent for TA0CP!");
    	Intent intent = new Intent();
        intent.setAction("ShuttleUnmount");
        intent.putExtra("devicePath", "/mnt/usb_storage"); 
        sendBroadcast(intent);
    }
    
    public boolean getUDiskDirs()
    {
    	File f;
    	
    	if (Build.VERSION.SDK_INT == 17)
    		f = new File("/mnt/sdcard/external_storage/");
    	else
    		f = new File("/mnt/");
    	
		if (f.exists())
		{
			File [] list = f.listFiles();
			for (int i=0; i< list.length; i++)
			{
				Log.e("U Disk", "Files: " + list[i].getName());
				if (list[i].isDirectory() && (	list[i].getName().toString().startsWith("sda") 
											 || list[i].getName().toString().startsWith("sdb")
											 /* Here add for Android 4.1 support */
											 || list[i].getName().toString().startsWith("usb")))
				{
					uList.add(list[i].getAbsolutePath().toString());
				}
			}
		}
		else
		{
			return false;
		}
		
		if (uList.size() <= 0)
		{
			return false;
		}else{
			return true;
		}
    }
    
    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        super.onCreateOptionsMenu(menu);
        menu.add(0, 0, 0, "Filemanager");        
        return true;
    }
    
    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        super.onOptionsItemSelected(item);
        switch(item.getItemId())
        {
    		case 0:
    			try{
    				Intent i = new Intent();
   					i.setClassName("com.oem.iFileManager", "com.oem.iFileManager.iFileManager");
    		    	startActivity(i);
    		    	bPass.setEnabled(true);
    		    	bFail.setEnabled(true);
    			}catch(ActivityNotFoundException e)
    			{
    				try{
    					Intent i = new Intent();
    					i.setClassName("com.fb.FileBrower", "com.fb.FileBrower.FileBrower");
    					startActivity(i);
        		    	bFail.setEnabled(true);
        		    	bPass.setEnabled(true);
    				}catch(ActivityNotFoundException e1){
    					Toast.makeText(getApplicationContext(),
    							"Open Filemanager Failed!",
    							Toast.LENGTH_SHORT)
    							.show();
    				}
    			}
    			return true;
    	}
        return true;
    }
	
	@Override
	public boolean onKeyDown(int keyCode, KeyEvent event) {
		if (keyCode == KeyEvent.KEYCODE_BACK) {
			return true;
		}
		return super.onKeyDown(keyCode, event);
	}
	
}